### ISSUE TITLE
Install winget into allusers vs. userprofile for the ability to run winget as SYSTEM

### ISSUE THREAD

**Issue by michaeldeblok** [microsoft/winget-cli#1627]
### Description of the new feature / enhancement

Currently, winget can only be installed under the current user and needs to be reinstalled each and every single time for each new user on their endpoint.

From a configuration management tool perspective, I'd like to see that winget is installed into %programdata% instead of %localappdata%. Benefits this will have are that it won't need to be installed for each user, and as admins, we can run remote PowerShell scripts to execute an installation or use a non-Microsoft configuration management tool that runs under the system account to install.

Additionally, some packages completely ignore the --silent switch and still bring up an interactive window alerting the user of an upgrade/install when we don't want this, running under a system account circumvents this issue.

### Proposed technical implementation details

From a configuration management tool perspective, I'd like to see that winget is installed into %programdata% instead of %localappdata%.

**Comment by woter1832** [comment]
So one cannot use winget for creating AVD/Sysprep's images!

```
2024-04-14 15:17:03, Error                 SYSPRP Package Microsoft.Winget.Source_2024.414.1454.918_neutral__8wekyb3d8bbwe was installed for a user, but not provisioned for all users. This package will not function properly in the sysprep image.
2024-04-14 15:17:03, Error                 SYSPRP Failed to remove apps for the current user: 0x80073cf2.
2024-04-14 15:17:03, Error                 SYSPRP Exit code of RemoveAllApps thread was 0x3cf2.
```

A bit of an epic fail, wouldn't you say @Microsoft?

**Comment by malfins** [comment]
Hm, has the sysprep problem been confirmed? Can't really find any information. This makes it really difficult/impossible to use Image Builder for 2025.

**Comment by ncook-hxgn** [comment]
Yup, sysprep confirmed. I thought this was going to be able to replace the ansible provisioner in my packerfile but it's going to be a murder scene.

> A bit of an epic fail, wouldn't you say [@microsoft](https://github.com/microsoft)?

Personal opinion - I tend to agree. Imagine being Microsoft and letting Red Hat set up a picnic table in yer nice walled garden. 

---

Edit: So, the symptom is that sysprep never finishes. It prints `IMAGE_STATE_COMPLETE`, but never reaches the `IMAGE_STATE_GENERALIZE_RESEAL_TO_OOBE` state.

I got some logs. I had to knobble the traditional sysprep script found in the packer docs to do it, basically to detect and break out of the inifinitely running loop:

```pwsh
$SysPrepActLog = "$Env:WINDIR\System32\Sysprep\Panther\setupact.log"
$SysPrepErrLog = "$Env:WINDIR\System32\Sysprep\Panther\setuperr.log"
$TooLongToWait = 600 # 10 minutes
$Waited = 0
# If Guest Agent services are installed, make sure that they have started.
foreach ($service in Get-Service -Name RdAgent, WindowsAzureTelemetryService, WindowsAzureGuestAgent -ErrorAction SilentlyContinue) { 
    while ((Get-Service $service.Name).Status -ne 'Running') {
        Start-Sleep -s 5
        $Waited += 5
    }
}

& $env:SystemRoot\\System32\\Sysprep\\Sysprep.exe /oobe /generalize /quiet /quit /mode:vm
while($true) {
    $imageState = Get-ItemProperty HKLM:\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Setup\\State | Select ImageState
    Write-Output $imageState.ImageState
    
    if($imageState.ImageState -eq 'IMAGE_STATE_GENERALIZE_RESEAL_TO_OOBE') {
        break
    }
    Start-Sleep -s 10
    $Waited += 10
    if ($Waited -ge $TooLongToWait) {
        Write-Host "Sysprep did not complete in a timely manner. Exiting."
        break
    }
}

if (Test-Path -Path $SysPrepActLog) {
    Write-Host "Sysprep completed successfully. Review the setupact.log for details."
    Get-Content -Path $SysPrepActLog | Out-Host
} else {
    Write-Host "Sysprep completed but setupact.log not found. This may indicate an issue."
}

if (Test-Path -Path $SysPrepErrLog) {
    Write-Host "Sysprep encountered errors. Review the setuperr.log for details."
    Get-Content -Path $SysPrepErrLog | Out-Host
} else {
    Write-Host "No errors logged in setuperr.log."
}
```

Sysprep error logs:

(note: `Package Microsoft.Winget.Source_2025.718.1153.54_neutral__8wekyb3d8bbwe was installed for a user, but not provisioned for all users. This package will not function properly in the sysprep image.`)

```log
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: Sysprep encountered errors. Review the setuperr.log for details.
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-06-07 09:52:50, Error                 SYSPRP BCD: BiUpdateEfiEntry failed c000000d
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-06-07 09:52:50, Error                 SYSPRP BCD: BiExportBcdObjects failed c000000d
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-06-07 09:52:50, Error                 SYSPRP BCD: BiExportStoreAlterationsToEfi failed c000000d
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-06-07 09:52:50, Error                 SYSPRP BCD: Failed to export alterations to firmware. Status: c000000d
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-06-17 12:41:38, Error                 SYSPRP BCD: BiUpdateEfiEntry failed c000000d
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-06-17 12:41:38, Error                 SYSPRP BCD: BiExportBcdObjects failed c000000d
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-06-17 12:41:38, Error                 SYSPRP BCD: BiExportStoreAlterationsToEfi failed c000000d
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-06-17 12:41:38, Error                 SYSPRP BCD: Failed to export alterations to firmware. Status: c000000d
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-06-17 12:41:41, Error                 SYSPRP GeneralizeForImaging for WaaSMedic failed to create WaaSRemediationAgent. hr = 0x80070422[gle=0x00000057]
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-07-18 11:33:41, Error                 SYSPRP Package Microsoft.Winget.Source_2025.718.1153.54_neutral__8wekyb3d8bbwe was installed for a user, but not provisioned for all users. This package will not function properly in the sysprep image.
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-07-18 11:33:41, Error                 SYSPRP Failed to remove apps for the current user: 0x80073cf2.
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-07-18 11:33:41, Error                 SYSPRP Exit code of RemoveAllApps thread was 0x3cf2.
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-07-18 11:33:41, Error                 SYSPRP ActionPlatform::LaunchModule: Failure occurred while executing 'SysprepGeneralizeValidate' from C:\Windows\System32\AppxSysprep.dll; dwRet = 0x3cf2
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-07-18 11:33:41, Error                 SYSPRP SysprepSession::Validate: Error in validating actions from C:\Windows\System32\Sysprep\ActionFiles\Generalize.xml; dwRet = 0x3cf2
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-07-18 11:33:41, Error                 SYSPRP RunPlatformActions:Failed while validating Sysprep session actions; dwRet = 0x3cf2
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-07-18 11:33:41, Error      [0x0f0070] SYSPRP RunDlls:An error occurred while running registry sysprep DLLs, halting sysprep execution. dwRet = 0x3cf2
2025-07-18T11:43:44Z: ==> windows-custom.azure-arm.windows-custom: 2025-07-18 11:33:41, Error      [0x0f00d8] SYSPRP WinMain:Hit failure while pre-validate sysprep generalize internal providers; hr = 0x80073cf2
```

**Comment by FDBTech** [comment]
I wish I came across this sooner. Windows 11 Enterprise LTSC does not include WinGet or the Microsoft store for that matter. I convinced my team to start using WinGet as our new package manager for Windows. WinGet is already included in all our other image types so the last thing to do was include it our base Windows 11 LTSC  image. Installing it was a pain if you're doing it through a Packer provisioner and then of course you go to sysprep the image...
